cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/ReadThroughBookCache.java

```diff
@@ -82,16 +86,20 @@ public BookInfo loadBook (BookId bookId) throws IOException {
         try {
             BookEntity bookEntity = ops.getCradleBookOperator().get(bookId.getName(), readAttrs);
 
+            if (bookEntity == null) {
+                throw new IOException();
+            }
+
             if (!bookEntity.getSchemaVersion().equals(schemaVersion)) {
-                throw new IOException(String.format("Unsupported schema version for the book \"%s\". Expected: %s, found: %s. Skipping",
+                throw new Exception(String.format(UNSUPPORTED_SCHEMA_VERSION_FORMAT,
                         bookEntity.getName(),
                         schemaVersion,
                         bookEntity.getSchemaVersion()));
             }
 
             return processBookEntity(bookEntity);
         } catch (Exception e) {
-            throw new IOException("Error while loading book", e);
+            throw new IOException(String.format("Error while loading book \"%s\"", bookId.getName()), e);
```

Comment:
If you use `CradleStorageException` when schema doesn't match, don't wrap it with `IOException`, because it is not related to I/O. Throw it as is.